pkg/controller/configuration/watcher.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configuration
+
+import (
+	"context"
+	"sync"
+
+	configapi "github.com/onosproject/onos-api/go/onos/config/v2"
+
+	"github.com/onosproject/onos-config/pkg/store/configuration"
+
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-config/pkg/store/topo"
+	"github.com/onosproject/onos-lib-go/pkg/controller"
+)
+
+const queueSize = 100
+
+// Watcher configuration store watcher
+type Watcher struct {
+	configurations configuration.Store
+	cancel         context.CancelFunc
+	mu             sync.Mutex
+}
+
+// Start starts the watcher
+func (w *Watcher) Start(ch chan<- controller.ID) error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if w.cancel != nil {
+		return nil
+	}
+
+	eventCh := make(chan configapi.ConfigurationEvent, queueSize)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	err := w.configurations.Watch(ctx, eventCh, nil)
+	if err != nil {
+		cancel()
+		return err
+	}
+	w.cancel = cancel
+	go func() {
+		for event := range eventCh {
+			ch <- controller.NewID(event.Configuration.ID)
+		}
+	}()
+	return nil
+}
+
+// Stop stops the watcher
+func (w *Watcher) Stop() {
+	w.mu.Lock()
+	if w.cancel != nil {
+		w.cancel()
+		w.cancel = nil
+	}
+	w.mu.Unlock()
+}
+
+// TopoWatcher topology watcher
+type TopoWatcher struct {
+	topo           topo.Store
+	configurations configuration.Store
+	cancel         context.CancelFunc
+	mu             sync.Mutex
+}
+
+// Start starts the topo store watcher
+func (w *TopoWatcher) Start(ch chan<- controller.ID) error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if w.cancel != nil {
+		return nil
+	}
+
+	eventCh := make(chan topoapi.Event, queueSize)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	err := w.topo.Watch(ctx, eventCh, nil)
+	if err != nil {
+		cancel()
+		return err
+	}
+	w.cancel = cancel
+	go func() {
+		for event := range eventCh {
+			if _, ok := event.Object.Obj.(*topoapi.Object_Entity); ok {
+				err = event.Object.GetAspect(&topoapi.Configurable{})
+				if err == nil {
+					configurations, err := w.configurations.List(ctx)
+					if err != nil {
+						log.Warn(err)
+						continue
+					}
+					for _, config := range configurations {
```

Comment:
Here If we use any IDs for configuration (i.e. not target ID), we need to find all the configuration that is associated with the target ID of the event.